### PR TITLE
Added matrix creation capabilities

### PR DESCRIPTION
**Pull request by smkch95**
**Explain fix**
This PR contains following code changes:
1. Definition of `Matrix` struct which holds the 2D data,
2. its creation from either in memory 2D source data or an IO reader that reads data from file source. Currently only `csv` formatted data is supported.

**Validation notes**
No testing/validations were done for this code change.

### CODE REVIEW

File: matrix.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	if row == 0 {
+		return nil, fmt.Errorf("Can not create matrix from empty data")
+	}
+	col := len(d[0])
+	data := make([][]float64, row)
+	for i := 0; i < row; i++ {
+		if len(d[i]) != col {
+			return nil, fmt.Errorf("Invalid data - not all rows are of same length")
+		}
+		data[i] = make([]float64, col)
```
Comment (sujoy0306) [inline matrix.go]: make is for typecasting?

File: matrix.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+		for j := 0; j < col; j++ {
+			data[i][j] = d[i][j]
+		}
+	}
+	return &Matrix{
+		row,
+		col,
+		data,
+	}, nil
```
Comment (sujoy0306) [inline matrix.go]: what's the significance of nil

File: matrix.go (hunk 1 above)
Comment (sujoy0306) [reply matrix.go]: ok

File: rw.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
+package matply
+
```
Comment (sujoy0306) [inline rw.go]: to run this code do I need to write the main every time?

Comment (smkch95) [reply rw.go]: This is a library. Just like you import `java.util.*` for your Java application, the user of this library would also import this and use it in its application. Yes, to run a Go application you need `package main` and `func main` as the entry point.

**Pull request merged**